src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs

```diff
@@ -45,11 +64,25 @@ public async Task<GetResult> GetDocumentById(IGetRequest getRequest)
 
         try
         {
-            return await _getDocumentById.GetById(getRequest);
+            await using var connection = await _dataSource.OpenConnectionAsync();
+            await using var transaction = await connection.BeginTransactionAsync(IsolationLevel.Serializable);
```

Comment:
Connections and transactions are now managed at this level. Enables transaction testability.

Reply:
As an aside, these are similar across the operations but pretty simple. I played with extracting commonality, but the result was awkward and I didn't like it.